cpan/flatpak-cpan-generator.pl

```diff
@@ -122,7 +122,7 @@ sub main {
       "    mod_dir=\$1",
       "    cd \$mod_dir",
       "    if [ -f 'Makefile.PL' ]; then",
-      "        perl Makefile.PL && make install",
+      "        perl Makefile.PL PREFIX=/ && make install DESTDIR=/app",
```

Comment:
This sounds wrong. PREFIX would be `/app`, DESTDIR should be unset.

Reply:
Technically PREFIX is `/usr` if its a runtime and `/app` for applications, this script can, although probably isn't, usable for both.

Reply:
It is counter intuitive, but this is what I get:

> Installing /app/lib/perl5/site_perl/5.34.0/x86_64-linux/auto/Compress/Raw/Lzma/Lzma.so
Installing /app/lib/perl5/site_perl/5.34.0/x86_64-linux/auto/Compress/Raw/Lzma/autosplit.ix
Installing /app/lib/perl5/site_perl/5.34.0/x86_64-linux/Compress/Raw/Lzma.pm
Installing /app/share/man/man3/Compress::Raw::Lzma.3
Appending installation info to /app//lib/perl5/5.34.0/x86_64-linux/perllocal.pod

Working test case https://github.com/Mailaender/flathub/tree/010editor

Reply:
I'm not a perl person but here is how these variables work in everything unix I've ever used.

- PREFIX is the actual prefix of the runtime environment, it is often passed to actual source code and is very relevant.
- DESTDIR is a prefix to copy files into, it is used only at install time and the purpose of it is for intermediate installation steps before final packaging. This does not apply to flatpak.

So while its possible Perl is doing something very odd. This just looks incorrect and risks the wrong PREFIX being used.

Reply:
What do you suggest instead to write into the correct directories?

Reply:
`PREFIX=${FLATPAK_DEST}`

keep `DESTDIR` unset

Reply:
Are you setting prefix during both build and install?

Reply:
That's it! Confirmed it works.